src/sparseml/evaluation/integrations/lm_eval_harness.py

```diff
@@ -0,0 +1,214 @@
+        tasks=task_names,
+        batch_size=batch_size,
+        device=device,
+        **kwargs,
+    )
+
+    results_raw = evaluator.simple_evaluate(**evaluator_input.dict())
+
+    results = Result(
+        raw=dict(output=results_raw, input=_filter_evaluator_input(evaluator_input)),
+        formatted=_format_lm_eval_raw_results(results_raw),
+    )
+
+    return results
+
+
+class LMEvalHarnessEvaluatorInputSchema(BaseModel):
+    model: Any = Field(description="The name of the model.")
+    tasks: List[str] = Field(
+        description="The task (or multiple tasks) to evaluate the target on."
+    )
+    batch_size: int = Field(description="The batch size to use for evaluation.")
+    model_args: str = Field(
+        "", description="Additional arguments for the evaluated model."
+    )
+    num_fewshot: int = Field(0, description="The number of few shots to use.")
+    max_batch_size: Optional[int] = Field(
+        None, description="Maximal batch size to try with --batch_size auto."
+    )
+    device: Optional[str] = Field(None, description="Device to use for evaluation.")
+    use_cache: Optional[str] = Field(
+        None,
+        description="A path to a sqlite db "
+        "file for caching model responses. `None` if not caching.",
+    )
+    limit: Optional[float] = Field(
+        None,
+        description="Limit the number of examples per task. If <1, "
+        "limit is a percentage of the total number of "
+        "examples.",
+    )
+    decontamination_ngrams_path: Optional[str] = Field(
+        None, description="Specify the path for decontamination n-grams."
+    )
+    check_integrity: bool = Field(
+        False, description="Include this flag to check integrity."
+    )
+    write_out: bool = Field(False, description="Include this flag to write out.")
+
+
+class SparseMLLM(HFLM):
+    """
+    SparseML is an open-source model optimization toolkit that enables you to create
+    inference-optimized sparse models using pruning, quantization, and distillation
+    algorithms. Models optimized with SparseML can then be exported to the ONNX and
+    deployed with DeepSparse for GPU-class performance on CPU hardware.
+
+    This class is a wrapper around the HuggingFace LM class to enable SparseML
+    integration with the lm-evaluation-harness
+    """
+
+    def _create_model(
+        self,
+        pretrained: str,
+        **kwargs,
+    ) -> None:
+        recipe_path = fetch_recipe_path(target=pretrained)
+        model_kwargs = kwargs if kwargs else {}
+        ignored_kwargs = [
+            "dtype",
+            "parallelize",
+            "device_map_option",
+            "max_memory_per_gpu",
+            "max_cpu_memory",
+            "peft",
+            "autogptq",
+        ]
+        for ignored_kwarg in ignored_kwargs:
+            model_kwargs.pop(ignored_kwarg, None)
+
+        device = model_kwargs.pop("device", None)
+        task = model_kwargs.pop("task", "text-generation")
```

Comment:
it will probably be less prone to upstream breaking changes if we explicitly define all the named args we want to support, rather than ignoring the kwargs we currently know to ignore

Reply:
Done!